rcnn/retrain_cnn.py

```diff
@@ -145,48 +148,43 @@ def create_image_lists(videos_dir, testing_percentage, validation_percentage):
     A dictionary containing an entry for each label subfolder, with images split
     into training, testing, and validation sets within each label.
   """
-  if not gfile.Exists(video_dir):
-    tf.logging.error("Video directory '" + video_dir + "' not found.")
-    return None
+
   result = {}
-  sub_dirs = [x[0] for x in gfile.Walk(video_dir)]
-  # The root directory comes first, so skip it.
-  is_root_dir = True
-  for sub_dir in sub_dirs:
-    if is_root_dir:
-      is_root_dir = False
-      continue
-    extensions = ['jpg', 'jpeg', 'JPG', 'JPEG']
-    file_list = []
-    dir_name = os.path.basename(sub_dir)
-    if dir_name == image_dir:
-      continue
-    tf.logging.info("Looking for images in '" + dir_name + "'")
-    for extension in extensions:
-      file_glob = os.path.join(image_dir, dir_name, '*.' + extension)
-      file_list.extend(gfile.Glob(file_glob))
-    if not file_list:
-      tf.logging.warning('No files found')
-      continue
-    if len(file_list) < 20:
+
+  for folder, labels in class_per_frame.items():
+    training_images = []
+    testing_images = []
+    validation_images = []
+
+    dir_name = os.path.join(videos_dir, folder)
+
+    images = sorted(glob.glob(os.path.join(dir_name, '*')))
+
+    if len(images) < 20:
       tf.logging.warning(
-          'WARNING: Folder has less than 20 images, which may cause issues.')
-    elif len(file_list) > MAX_NUM_IMAGES_PER_CLASS:
+          'WARNING: Folder {} has less than 20 images, which may cause issues.').format(dir_name)
+    elif len(images) > MAX_NUM_IMAGES_PER_CLASS:
       tf.logging.warning(
           'WARNING: Folder {} has more than {} images. Some images will '
           'never be selected.'.format(dir_name, MAX_NUM_IMAGES_PER_CLASS))
-    label_name = re.sub(r'[^a-z0-9]+', ' ', dir_name.lower())
-    training_images = []
-    testing_images = []
-    validation_images = []
-    for file_name in file_list:
-      base_name = os.path.basename(file_name)
+
+    for image_path in images:
+      timestamp = image_path.replace('.jpg', '').split('/')[-1] # This will brake in Windows I think, any ideas?
```

Comment:
Check in os, I think there are ways of doing this. Like os.basename or basefilename, something like that.